src/plugins/typechecker/testmod_typechecker.c

```diff
@@ -0,0 +1,116 @@
+#define KEY3_NAME PARENT_KEY_NAME "/key3"
+#define KEY4_NAME PARENT_KEY_NAME "/key4"
+
+static KeySet * keysetWithStandardFunctions (void)
+{
+	KeySet * ks = ksNew (0, KS_END);
+
+	ksAppendKey (ks, keyNew (PARENT_KEY_NAME "/elektra/spec/fallback/#", KEY_META, "elektra/spec/type",
+				 "RegexContains b a => Key b :: . -> Key a -> Key a", KEY_META, "elektra/spec/impl",
+				 "fallback a (Key P.Nothing) = a \n fallback _ a = a", KEY_END));
+
+	ksAppendKey (ks, keyNew (PARENT_KEY_NAME "/elektra/spec/override/#", KEY_META, "elektra/spec/type",
+				 "RegexContains b a => Key b :: . -> Key a -> Key a", KEY_META, "elektra/spec/impl",
+				 "override (Key P.Nothing) b = b \n override a _ = a", KEY_END));
+
+	ksAppendKey (ks, keyNew (PARENT_KEY_NAME "/elektra/spec/check/range", KEY_META, "elektra/spec/type",
+				 "RegexIntersects a b => P.Proxy b :: Range . -> Key a -> Key (RegexIntersection a b)", KEY_META,
+				 "elektra/spec/impl", "checkrange _ a = a", KEY_END));
+
+	ksAppendKey (ks, keyNew (PARENT_KEY_NAME "/elektra/spec/check/long", KEY_META, "elektra/spec/type",
+				 "RegexIntersects a \"" LONG_REGEX "\" => Key a -> Key (RegexIntersection a \"" LONG_REGEX "\")", KEY_META,
+				 "elektra/spec/impl", "checklong a = a", KEY_END));
+
+	return ks;
+}
+
+static void test_basics (void)
+{
+	printf ("test basics\n");
+
+	Key * parentKey = keyNew (PARENT_KEY_NAME, KEY_END);
+	KeySet * conf = ksNew (0, KS_END);
+	PLUGIN_OPEN ("typechecker");
+
+	KeySet * ks = keysetWithStandardFunctions ();
+
+	ksAppendKey (ks, keyNew (KEY1_NAME, KEY_VALUE, "2500", KEY_META, "check/range", "0-5000", KEY_END));
+
+	ksAppendKey (ks, keyNew (KEY2_NAME, KEY_VALUE, "500", KEY_META, "fallback/#1", KEY1_NAME, KEY_META, "check/range", "-250-7500",
+				 KEY_END));
+
+	ksAppendKey (ks, keyNew (KEY3_NAME, KEY_META, "fallback/#1", KEY1_NAME, KEY_META, "fallback/#2", KEY2_NAME, KEY_END));
+
+	ksAppendKey (ks, keyNew (KEY4_NAME, KEY_META, "fallback/#1", KEY1_NAME, KEY_META, "override/#1", KEY2_NAME, KEY_META, "check/long",
+				 "", KEY_END));
+
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_NO_UPDATE, "call to kdbSet was not successful");
+	succeed_if (output_error (parentKey), "error(s) found but none expected");
+
+	keyDel (parentKey);
+	ksDel (ks);
+	PLUGIN_CLOSE ();
+}
+
+static void test_invalid_ranges_override (void)
+{
+	printf ("test invalid ranges fallback\n");
+
+	Key * parentKey = keyNew (PARENT_KEY_NAME, KEY_END);
+	KeySet * conf = ksNew (0, KS_END);
+	PLUGIN_OPEN ("typechecker");
+
+	KeySet * ks = keysetWithStandardFunctions ();
+
+	ksAppendKey (ks, keyNew (KEY1_NAME, KEY_VALUE, "2500", KEY_META, "check/range", "0-5000", KEY_END));
+
+	ksAppendKey (ks, keyNew (KEY2_NAME, KEY_VALUE, "500", KEY_META, "override/#1", KEY1_NAME, KEY_META, "check/range", "7200-10000",
+				 KEY_END));
+
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_NO_UPDATE, "call to kdbSet was not successful");
```

Comment:
Okay, the "setmeta" bug above seems to be the wrong return value here. You need to return ELEKTRA_PLUGIN_STATUS_ERROR.

Reply:
fixed